src/Platform.Handlers/src/Xamarin.Platform.Handlers/Hosting/Internal/AppHostLifetime.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Xamarin.Platform.Hosting.Internal
+{
+	public class AppHostLifetime : IHostLifetime, IDisposable
```

Comment:
Does this class need to be `public`? It is in an `Internal` namespace, which seems odd.

I know that's how Microsoft.Extensions.Hosting is, but I'm not sure if that was intentional, or a mistake.